File: client2/src/game/components/position.ts
import { Flag } from './flags'

export class Position {
  static _id = 'Position'
  _id = Position._id

  static _flag = Flag.Position
  _flag = Flag.Position

  x: number
  y: number

  constructor(x: number = 0, y: number = 0) {
    this.x = x
    this.y = y
  }
}

File: packages/client2/lib/game/components/position.dart
import '../../ecs/ecs.dart';
import '../flags.dart';

class Position extends Component {
  static const int Flag = ComponentFlags.Position;
  int flag = Position.Flag;

  static const String Id = 'Position';
  String id = Position.Id;

  double x = 0;
  double y = 0;

  Position({
    this.x,
    this.y,
  });

  @override
  String toString() {
    return 'Position (x: $x, y: $y)';
  }
}
